Source/Core/DolphinQt2/Config/DockWidget.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt2/Config/DockWidget.h"
+
+#include <QEvent>
+#include <QSettings>
+#include <QTabWidget>
+
+#include "DolphinQt2/Config/LogConfigWidget.h"
+#include "DolphinQt2/Config/LogWidget.h"
+#include "DolphinQt2/Settings.h"
+
+DockWidget::DockWidget(QWidget* parent) : QDockWidget(parent)
+{
+  setAllowedAreas(Qt::AllDockWidgetAreas);
+
+  CreateWidgets();
+  ConnectWidgets();
+
+  OnTabVisibilityChanged();
+
+  QSettings settings;
+
+  restoreGeometry(settings.value(QStringLiteral("dock/geometry")).toByteArray());
+  setFloating(settings.value(QStringLiteral("dock/floating")).toBool());
+}
+
+void DockWidget::CreateWidgets()
+{
+  m_tab_widget = new QTabWidget;
+
+  m_log_widget = new LogWidget;
+  m_log_config_widget = new LogConfigWidget;
+
+  setWidget(m_tab_widget);
+}
+
+void DockWidget::ConnectWidgets()
+{
+  connect(this, &QDockWidget::topLevelChanged, this, &DockWidget::SaveSettings);
+  connect(&Settings::Instance(), &Settings::LogVisibilityChanged, this,
+          &DockWidget::OnTabVisibilityChanged);
+  connect(&Settings::Instance(), &Settings::LogConfigVisibilityChanged, this,
+          &DockWidget::OnTabVisibilityChanged);
+}
+
+void DockWidget::OnTabVisibilityChanged()
+{
+  m_tab_widget->clear();
+
+  if (Settings::Instance().IsLogVisible())
+    m_tab_widget->addTab(m_log_widget, tr("Log"));
+
+  if (Settings::Instance().IsLogConfigVisible())
+    m_tab_widget->addTab(m_log_config_widget, tr("Log Configuration"));
+
+  setHidden(m_tab_widget->count() == 0);
+}
+
+bool DockWidget::eventFilter(QObject* object, QEvent* event)
```

Comment:
This `eventFilter` function never gets called. Did you mean to override `event(QEvent*)`?